lib/elixir/lib/code/normalizer.ex

```diff
@@ -0,0 +1,246 @@
+defmodule Code.Normalizer do
+  @moduledoc false
+
+  defguard is_literal(x)
+           when is_integer(x) or
+                  is_float(x) or
+                  is_binary(x) or
+                  is_atom(x)
+
+  @doc """
+  Wraps literals in the quoted expression to conform to the AST format expected
+  by the formatter.
+  """
+  def normalize(quoted, opts \\ []) do
+    line = Keyword.get(opts, :line, 1)
+
+    do_normalize(quoted, line: line)
+  end
+
+  # Skip already wrapped blocks
+  defp do_normalize({:__block__, _, [literal]} = quoted, _parent_meta) when is_literal(literal) do
+    quoted
+  end
+
+  defp do_normalize({:__block__, meta, args} = quoted, _parent_meta) do
+    if Keyword.has_key?(meta, :format) do
+      quoted
+    else
+      {:__block__, meta, Enum.map(args, &do_normalize(&1, meta))}
+    end
+  end
+
+  # Skip aliases so the module segment atoms don't get wrapped
+  defp do_normalize({:__aliases__, _, _} = quoted, _parent_meta) do
```

Comment:
You need to traverse the first element of __aliases__ in case it is not an atom. For example:

    __MODULE__.Bar.Baz

Maybe __MODULE__ will be fine but the first element of __aliases__ can be any expression.

Reply:
Done!